context/src/main/java/io/micronaut/runtime/converters/time/TimeConverterRegistrar.java

```diff
@@ -208,21 +244,63 @@ public void register(ConversionService<?> conversionService) {
                 OffsetDateTime.class,
                 (object, targetType, context) -> {
                     try {
-                        DateTimeFormatter formatter = resolveFormatter(context);
-                        OffsetDateTime result = OffsetDateTime.parse(object, formatter);
-                        return Optional.of(result);
+                        return format(context,
+                                object,
+                                OffsetDateTime::parse,
+                                OffsetDateTime::parse);
                     } catch (DateTimeParseException e) {
                         context.reject(object, e);
                         return Optional.empty();
                     }
                 }
         );
+
+        // CharSequence -> OffsetTime
+        conversionService.addConverter(
+                CharSequence.class,
+                OffsetTime.class,
+                (object, targetType, context) -> {
+                    try {
+                        return format(context,
+                                object,
+                                OffsetTime::parse,
+                                OffsetTime::parse);
+                    } catch (DateTimeParseException e) {
+                        context.reject(object, e);
+                        return Optional.empty();
+                    }
+                }
+        );
+    }
+
+    private <Y, T> Optional<Y> format(ConversionContext context,
+                                      T value,
+                                      BiFunction<T, DateTimeFormatter, Y> formatDate,
+                                      Function<T, Y> defaultFormat) {
+        return Optional.of(context.getFormat().map(pattern -> {
+            if (ConversionContext.RFC_1123_FORMAT.equals(pattern)) {
+                T converted = value;
+                if (converted instanceof TemporalAccessor) {
+                    converted = (T) toGmt((TemporalAccessor) converted);
+                }
+                return formatDate.apply(converted, DateTimeFormatter.RFC_1123_DATE_TIME);
+            } else {
+                return formatDate.apply(value, DateTimeFormatter.ofPattern(pattern, context.getLocale()));
```

Comment:
Is it normal to recreate formatter for every field parse? Maybe it's better to save the already created vormatters in some kind of map?

Reply:
@graemerocher ?